### PR TITLE
DOC Document official support for linkfield

### PR DESCRIPTION
**Pull request by GuySartorelli**
## Description

Documents providing official support for `silverstripe/linkfield` 4.0.0

Note that this PR assumes the related PR is merged to pull linkfield docs from the linkfield repository.
The renamed files are so that a new "Optional Features" section can be inserted between "developer guides" and "upgrading".
The "optional features" section will be effectively the same as the userhelp docs in that it will pull content from other repositories (for the scope of the current work, just linkfield).

## Issues

- https://github.com/silverstripe/silverstripe-linkfield/issues/27

## Pull request checklist

- [x] The target branch is correct
    - See [branches and commit messages](https://docs.silverstripe.org/en/contributing/documentation#branches-and-commit-messages)
- [x] All commits are relevant to the purpose of the PR (e.g. no TODO comments, unrelated rewording/restructuring, or arbitrary changes)
    - Small amounts of additional changes are usually okay, but if it makes it hard to concentrate on the relevant changes, ask for the unrelated changes to be reverted, and submitted as a separate PR.
- [x] The commit messages follow our [commit message guidelines](https://docs.silverstripe.org/en/contributing/code/#commit-messages)
- [x] The PR follows our [contribution guidelines](https://docs.silverstripe.org/en/contributing/documentation/)
- [x] The changes follow our [writing style guide](https://docs.silverstripe.org/en/contributing/documentation/#writing-style)
- [x] Code examples follow our [coding conventions](https://docs.silverstripe.org/en/contributing/coding_conventions/)
- [ ] CI is green

### CODE REVIEW

File: en/04_Optional_Features/index.md
Comment (ChloeSartorelli) [inline en/04_Optional_Features/index.md:1]: Based on https://github.com/silverstripe/silverstripe-userhelp-content/blob/5/docs/en/03_Optional_features/index.md

File: en/12_Project_Governance/07_Supported_Modules.md [hunk 1]
```diff
@@ -7,7 +7,8 @@ description: Modules which are commercially supported by Silverstripe
 
 Silverstripe CMS ships modules that receive commercial support. Commercially supported modules receive regular updates to work with the latest Silverstripe CMS release. Their [public API](/project_governance/public_api/) conforms to [semantic versioning](https://semver.org/). They are covered by:
 
-- our [security release process](/contributing/managing_security_issues) and
+- our [security release process](/contributing/managing_security_issues)
+- our [minor release policy](minor_release_policy)
 - our [major release policy](major_release_policy).
```
Comment (ChloeSartorelli) [inline en/12_Project_Governance/07_Supported_Modules.md:12]: This is an unrelated change - I noticed we don't call out the minor release policy here but we should.

File: en/04_Optional_Features/index.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+---
+title: Optional features
+---
+
+# Optional features
+
+Silverstripe CMS can be highly customised using both supported and community-made modules. You can find these [one Packagist](https://packagist.org/?type=silverstripe-vendormodule).
+
+This section highlights some of the CMS features provided by the [Silverstripe CMS supported modules](/project_governance/supported_modules/).
+In each case you'll need to ensure you have the module installed in your project in order to use the features described here.
```
Comment (emteknetnz) [inline en/04_Optional_Features/index.md:10]: Should either have a newline above this or have both sentences on one line

File: en/08_Changelogs/5.2.0.md [hunk 3]
```diff
@@ -35,6 +36,33 @@ We have provided a severity rating of the vulnerabilities below based on the CVS
...
 
+## Official support for `silverstripe/linkfield` {#linkfield}
+
+[`silverstripe/linkfield`](https://github.com/silverstripe/silverstripe-linkfield) 4.0.0 has been released and is now a commercially supported module.
+
+That means it is now covered by our [minor release policy](/project_governance/minor_release_policy/), our [major release policy](/project_governance/major_release_policy/), and our [security release process](/contributing/managing_security_issues).
+
+This module provides a [`Link`](api:SilverStripe\LinkField\Models\Link) model and two new form fields: [`LinkField`](api:SilverStripe\LinkField\Form\LinkField) and [`MultiLinkField`](api:SilverStripe\LinkField\Form\MultiLinkField). The intention of the module is to provide a clear and consistent approach to managing links in the CMS.
+
+This release of the module includes a bunch of features that were sorely missing from earlier versions including:
```
Comment (emteknetnz) [inline en/08_Changelogs/5.2.0.md]:
```suggestion
This release of the module includes a bunch of features that were missing from earlier versions including:
```

File: en/08_Changelogs/5.2.0.md [hunk 4]
```diff
@@ -35,6 +36,33 @@ We have provided a severity rating of the vulnerabilities below based on the CVS
...
+
+- support for `has_many` relations
+- versioned support
+- validation
+- the ability to declare which link types are allowed per field
+- improved UI including better keyboard navigation.
```
Comment (emteknetnz) [inline en/08_Changelogs/5.2.0.md]:
```suggestion
- Support for `has_many` relations
- Versioned support
- Validation
- The ability to declare which link types are allowed per field
- Improved UI including better keyboard navigation.
```

**Review by emteknetnz — CHANGES_REQUESTED** [review]

File: en/04_Optional_Features/index.md (hunk 2 above)
Comment (ChloeSartorelli) [reply en/04_Optional_Features/index.md:10]: We explicitly don't have a standard for whether to break up lines or not (this was one of the linting rules we opted *not* to enable). it doesn't affect anything.

File: en/08_Changelogs/5.2.0.md (hunk 3 above)
Comment (ChloeSartorelli) [reply en/08_Changelogs/5.2.0.md]: Done

File: en/08_Changelogs/5.2.0.md (hunk 4 above)
Comment (ChloeSartorelli) [reply en/08_Changelogs/5.2.0.md]: The old company style guide was very explicit about list styles but since the new one is less strict I'll make this change.

**Review by emteknetnz — APPROVED** [review]
